.github/workflows/lint.yml

```diff
@@ -8,17 +8,36 @@ on:
     branches:
       - master
 
+env:
+  GO_VERSION: stable
+  GOLANGCI_LINT_VERSION: v1.63
+
 jobs:
-  lint:
+  detect-modules:
     runs-on: ubuntu-latest
+    outputs:
+      modules: ${{ steps.set-modules.outputs.modules }}
     steps:
-      - uses: actions/setup-go@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
         with:
-          go-version: 1.22.x
+          go-version: ${{ env.GO_VERSION }}
+      - id: set-modules
+        run: echo "modules=$(go list -m -json | jq -s '.' | jq -c '[.[].Dir]')" >> $GITHUB_OUTPUT
 
+  lint:
+    needs: detect-modules
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        modules: ${{ fromJSON(needs.detect-modules.outputs.modules) }}
+    steps:
       - uses: actions/checkout@v4
-
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: golangci-lint ${{ matrix.modules }}
+        uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.56.2
+          version: ${{ env.GOLANGCI_LINT_VERSION }}
+          working-directory: ${{ matrix.modules }}
```

Comment:
I'm unsure to understand why you are computing all modules then run sequentially via the matrix

For me, as long as you have a go.work, `go test ./...` and `golangci-lint run ./...` should work in a transverse way

So for me, you don't need the `go list -json | jq`

Could you break something dbtests/mysql_test.go either with code and style

Then try to run go test and golangci-lint

They should be detected.

It's worth trying

Reply:
> For me, as long as you have a go.work, go test ./... and golangci-lint run ./... should work in a transverse way

Unfortunately it does not work in that way, you need to run `go test` in each `go.mod` directory, check this https://github.com/golang/go/issues/71294

Anyway I am still experimenting and I will try few different ideas to check, which one least sucks

ty for your input

Reply:
For anyone interested

> This proposal has been added to the [active column](https://go.dev/s/proposal-status#active) of the proposals project
and will now be reviewed at the weekly proposal review meetings.

_Originally posted by @rsc in https://github.com/golang/go/issues/71294#issuecomment-2637865401_

🤞

Reply:
> Unfortunately it does not work in that way, you need to run go test in each go.mod directory, check this golang/go#71294\n\nAnyway I am still experimenting and I will try few different ideas to check, which one least sucks

While I'm now convinced about the lack of feature in `go test`, I'm doubtful about the fact golangci-lint doesn't support it

A few weeks ago, I added a simple go.work file in gofr framework

https://github.com/gofr-dev/gofr/pull/1426

And all the errors appeared locally when I was running golangci-lint manually locally

Reply:
@ccoVeille here is a proof. I purposely entered some garbage to `dbtests` module, so lint should fail, but a simple `golangci-lint run` does not detect it

![image](https://github.com/user-attachments/assets/214ea21f-18f0-4550-902d-bad676e7dd7f)